python-squid/src/test/java/org/sonar/python/tree/PythonTreeMakerTest.java

```diff
@@ -1735,7 +1737,9 @@ public void generator_expression() {
     setRootRule(PythonGrammar.TEST);
     ComprehensionExpression generator = (ComprehensionExpression) parse("(x*x for x in range(10))", treeMaker::expression);
     assertThat(generator.getKind()).isEqualTo(Tree.Kind.GENERATOR_EXPR);
-    assertThat(generator.children()).hasSize(2);
+    assertThat(generator.children()).hasSize(4);
+    assertThat(((Token) generator.children().get(0)).value()).isEqualTo("(");
+    assertThat(((Token) generator.children().get(3)).value()).isEqualTo(")");
```

Comment:
I think this test can be more robust by checking that this token should be equal to "firstToken" and "lastToken"